src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Sse2.cs

```diff
@@ -22,67 +22,76 @@ public new abstract class X64 : Sse.X64
         {
             internal X64() { }
 
-            public new static bool IsSupported { get => IsSupported; }
+            public new static bool IsSupported { [Intrinsic]get => IsSupported; }
 
             /// <summary>
             /// __int64 _mm_cvtsd_si64 (__m128d a)
             ///   CVTSD2SI r64, xmm/m64
             /// This intrinisc is only available on 64-bit processes
             /// </summary>
+            [Intrinsic]
```

Comment:
Why do we need the `[Intrinsic]` attribute on everything now?

Reply:
This is a trade-off, we can do
1. Checking additional namespace and class name (`""` + `X64`) in the MethodTableBuilder
2. Or, adding `[Intrinsic]` for each X64 intrinsic

The first approach lets MethodTableBuilder checking additional namespace and class name on many candidates, so I choose the second one.

Reply:
I thought we already just said any method in `System.Runtime.Intrinsics.X86` is intrinsic?

https://github.com/dotnet/coreclr/blob/c94d8e68222d931d4bb1c4eb9a52b4b056e85f12/src/vm/methodtablebuilder.cpp#L1503

Reply:
But `X64` nested classes are not in `System.Runtime.Intrinsics.X86`...

Reply:
They are for the namespace, are they not?

Reply:
The runtime returns empty string for these X64 methods. But anyway, the methodtablebuilder code is an optimization for saving disk space, and we just have a few of `X64` methods,so not using this optimization is okay.

Let me update the comments in methodtablebuilder.

Reply:
How difficult would it be to just fix the methodtablebuilder instead?

We are currently just calling `GetNameOfTypeDef`, presumably we could change that to use the new overload you just added instead (would that add too much overhead, or is it effectively a nop if the type is not enclosed)?

Reply:
I agree that it would be better to fix the MethodTableBuilder here.

Reply:
Yes, if we have a vm API that can get the "actual" namespace, everything would be simpler.